contracts/governance_admin/src/query/query_refundable_funds.rs

```diff
@@ -0,0 +1,166 @@
+pub fn query_refundable_funds(
+    deps: Deps,
+    request_desciption: &GovAdminProposalRequestType,
+) -> ContractResult<RefundResponse> {
+    let (proposal_id, refund_receiver, user_total_deposits) = match request_desciption {
+        GovAdminProposalRequestType::PoolCreationRequest { request_id } => {
+            // query pool creation request
+            let pool_creation_request_context =
+                POOL_CREATION_REQUEST_DATA.load(deps.storage, *request_id)?;
+
+            let proposal_id =
+                pool_creation_request_context
+                    .status
+                    .proposal_id()
+                    .ok_or(ContractError::Std(StdError::generic_err(format!(
+                        "Proposal id not found for pool creation request id {}",
+                        request_id
+                    ))))?;
+
+            // validate that the funds are not claimed back already
+            let status = pool_creation_request_context.status;
+            match status {
+                PoolCreationRequestStatus::RequestFailedAndRefunded {
+                    proposal_id: _,
+                    refund_block_height,
+                } => {
+                    return Err(ContractError::Std(StdError::generic_err(format!(
+                        "Funds are already claimed back for this pool creation request in block {refund_block_height}",
+                    ))));
+                }
+                PoolCreationRequestStatus::RequestSuccessfulAndDepositRefunded {
+                    proposal_id: _,
+                    refund_block_height,
+                } => {
+                    return Err(ContractError::Std(StdError::generic_err(format!(
+                        "Funds are already claimed back for this pool creation request in block {refund_block_height}",
+                    ))));
+                }
+                _ => (),
+            }
+
+            (
+                proposal_id,
+                pool_creation_request_context.request_sender,
+                pool_creation_request_context.user_deposits_detailed,
+            )
+        }
+        GovAdminProposalRequestType::RewardSchedulesCreationRequest { request_id } => {
+            // query reward schedule creation request
+
+            let reward_schedule_request_state =
+                REWARD_SCHEDULE_REQUESTS.load(deps.storage, *request_id)?;
+
+            let proposal_id =
+                reward_schedule_request_state
+                    .status
+                    .proposal_id()
+                    .ok_or(ContractError::Std(StdError::generic_err(format!(
+                        "Proposal id not found for reward schedule creation request id {}",
+                        request_id
+                    ))))?;
+
+            // validate that the funds are not claimed back already
+            let status = reward_schedule_request_state.status;
+
+            if let RewardSchedulesCreationRequestStatus::RequestFailedAndRefunded {
```

Comment:
major: case for reclaim of deposit on success case is not handled. check above in pool creation request.

Reply:
Done